### PR TITLE
[chore] [receiver/elasticsearch] Use NewDefaultClientConfig instead of manual initialization

### PR DESCRIPTION
**Pull request by rogercoll**
**Description:** <Describe what has changed.>

Similar to https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35452 but for the `elasticsearch` receiver.

**Link to tracking Issue:** <Issue number if applicable> https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/35457

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by rogercoll** [comment]
Skip changelog?

File: receiver/elasticsearchreceiver/factory.go [hunk 1]
```diff
@@ -35,12 +35,18 @@ func createDefaultConfig() component.Config {
...
 
+	clientConfig := confighttp.NewDefaultClientConfig()
+	clientConfig.Endpoint = defaultEndpoint
+	clientConfig.Timeout = defaultHTTPClientTimeout
+	clientConfig.MaxIdleConns = nil
+	clientConfig.MaxIdleConnsPerHost = nil
+	clientConfig.MaxConnsPerHost = nil
+	clientConfig.IdleConnTimeout = nil
+	clientConfig.Headers = nil
+
```
Comment (mackjmr) [inline receiver/elasticsearchreceiver/factory.go:46]:
```
	clientConfig.MaxIdleConns = nil
	clientConfig.MaxIdleConnsPerHost = nil
	clientConfig.MaxConnsPerHost = nil
	clientConfig.IdleConnTimeout = nil
	clientConfig.Headers = nil
```

This part can be deleted, for MaxIdleConns, MaxIdleConnsPerHost, MaxConnsPerHost, IdleConnTimeout even if you set to nil here, the same defaults end up being used in the call to ToClient: https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L136.

For headers, ToClient checks for len (https://github.com/open-telemetry/opentelemetry-collector/blob/main/config/confighttp/confighttp.go#L210), so whether it's nil or an empty map (set in NewDefaultClientConfig) behaviour should be the same.

Comment (rogercoll) [reply receiver/elasticsearchreceiver/factory.go:41]: Good point, thanks for the detailed explanation. Removed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35509/commits/a9396e056d676e6ed2f746d9dc3d7ee50edd1b54

**Review by mackjmr — APPROVED** [review]

File: receiver/elasticsearchreceiver/config_test.go [hunk 2]
```diff
@@ -138,6 +140,7 @@ func TestValidateEndpoint(t *testing.T) {
 }
 
 func TestLoadConfig(t *testing.T) {
+	defaultTransport := http.DefaultTransport.(*http.Transport)
```
Comment (mackjmr) [inline receiver/elasticsearchreceiver/config_test.go:143]: One last thing, can you change this to match: https://github.com/open-telemetry/opentelemetry-collector/pull/11299/commits/aa03d7624950b58598e0fe1cbce27d408bf38424. Like that, we aren't affected if the global DefaultTransport is changed and define our own variables

Comment (rogercoll) [reply receiver/elasticsearchreceiver/config_test.go]: nice suggestion, changed in https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/35509/commits/506af7876458b038f29d20e4e5850ca837000deb

**Review by djaglowski — APPROVED** [review]

**Pull request merged**